tests/contrib/network/ovs/test_ovs.py

```diff
@@ -125,6 +125,29 @@ def test_add_bridge_port(self):
             mock.call(['ip', 'link', 'set', 'eth1', 'promisc', 'off'])
         ])
 
+    def test_del_bridge_port(self):
+        self.patch_object(ovs.subprocess, 'check_call')
+        self.patch_object(ovs, 'log')
+        ovs.del_bridge_port('test', 'eth1')
+        self.check_call.assert_has_calls([
+            mock.call(['ovs-vsctl', '--', '--if-exists', 'del-port',
+                       'test', 'eth1']),
+            mock.call(['ip', 'link', 'set', 'eth1', 'down']),
+            mock.call(['ip', 'link', 'set', 'eth1', 'promisc', 'off'])
+        ])
+        self.assertTrue(self.log.call_count == 1)
+        self.check_call.reset_mock()
+        ovs.del_bridge_port('test', 'eth1', promisc=None)
+        self.check_call.assert_has_calls([
```

Comment:
`assert_has_calls` tolerates extra calls, so you need an extra check to make sure there hasn't been other calls before and after

Reply:
I do check all the permutations and finally do a `assert_called_once` with a reset of the mock in between each round, so I think that should cover all bases.

Reply:
You're right, I think it's ok, but for the record this is what I mean:

```
    def test_del_bridge_port(self):
        self.patch_object(ovs.subprocess, 'check_call')
        self.patch_object(ovs, 'log')
        ovs.del_bridge_port('test', 'eth1')
        self.check_call.assert_has_calls([
            mock.call(['ovs-vsctl', '--', '--if-exists', 'del-port',
                       'test', 'eth1']),
            mock.call(['ip', 'link', 'set', 'eth1', 'down']),
            mock.call(['ip', 'link', 'set', 'eth1', 'promisc', 'off'])
        ])
        self.assertTrue(self.log.call_count == 1)

        # HERE IS WHAT I MEAN: at this point if check_call has also been called with other calls,
        # it will pass because you don't do (self.assertTrue(self.check_call.call_count == 3)

        self.check_call.reset_mock()
        ovs.del_bridge_port('test', 'eth1', linkdown=False)
        self.check_call.assert_called_once_with(
            ['ovs-vsctl', '--', '--if-exists', 'del-port', 'test', 'eth1'])
```